src/main/java/UseCases/CompatibilityFunction.java

```diff
@@ -0,0 +1,57 @@
+package UseCases;
+
+import Entities.User;
+import Entities.UserDataClasses.HideableUserDataClasses.Courses;
+import Entities.UserDataClasses.HideableUserDataClasses.Interests;
+import Entities.UserDataClasses.PrivateUserDataClasses.Preferences;
+import Entities.UserDataClasses.UserDataDictionaries.InterestsDict;
+
+public class CompatibilityFunction {
+
+    public float getCompatibility(User user1, User user2){
+        // Return 0 if two Users are incompatible (one user has an attribute
+        // which is a dealbreaker for the other), or 1 if they are compatible
+        float weight = 0;
+
+        Preferences user1Preferences = new Preferences();
+        Preferences user2Preferences = new Preferences();
+        user1Preferences.setData(user1.getPreferences().getData());
```

Comment:
This seems like a redundant object instantiation. Could you not just use the user preferences as the variable instead of setting it to a new Preferences object?

Reply:
That was what I did originally, but creating a Preferences object allowed me to use methods that I would not have access to otherwise, since User.getPreferences() returns a UserData object instead of a Preferences one. I should probably make it so that the getter functions return the respective object class, though.